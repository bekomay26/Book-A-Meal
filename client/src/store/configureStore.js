import { createStore, applyMiddleware } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/index';


const configureStore = initialState =>
  createStore(
    rootReducer,
    initialState,
    applyMiddleware(thunk, reduxImmutableStateInvariant()),
  );

export default configureStore;
