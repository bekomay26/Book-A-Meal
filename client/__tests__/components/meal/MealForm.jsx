import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import ConnectedMealForm, { MealForm } from '../../../src/components/meal/MealForm';
import { getMealsResponse } from '../../__mocks__/mockMeal';
import { getExtrasResponse } from '../../__mocks__/mockExtras';

// import jsdom from 'jsdom';
// const doc = jsdom.jsdom('<!doctype html><html><body></body></html>')
// global.document = doc
// global.window = doc.defaultView
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore({
  authReducer: {
    isAuthenticated: false,
    isCaterer: false,
    errors: {},
  },
});

const mealArray = getMealsResponse.meals;
const extraArray = getExtrasResponse.extras;

function setup() {
  const props = {
    meal: mealArray[0],
    onSave: jest.fn(),
    saving: false,
    imageUrl: '',
    cardImgList: [],
    handleCardChange: jest.fn(),
    onChange: jest.fn(),
    extrasList: ['meal', 'meal'],
    extrasTopList: ['meal', 'meal'],
    goes: extraArray,
    top: extraArray,
    addExtra: jest.fn(),
    selectChange: jest.fn(),
    selected: jest.fn(),
    extraOptId: 0,
  };
  return shallow(<ConnectedMealForm {...props} />, reactrouter.get());
}

function mountSetup() {
  const props = {
    meal: mealArray[0],
    onSave: jest.fn(),
    saving: false,
    imageUrl: '',
    cardImgList: [],
    handleCardChange: jest.fn(),
    onChange: jest.fn(),
    extrasList: [<select />],
    extrasTopList: ['meal', 'meal'],
    goes: [],
    top: [],
    addExtra: jest.fn(),
    selectChange: jest.fn(),
    selected: jest.fn(),
    extraOptId: 0,
  };
  return mount(<Provider store={store}><MealForm {...props} /></Provider>, reactrouter.get());
}

describe('The Meal Form Component', () => {
  // it('should render meal form component correctly', () => {
  //   const wrapper = setup();
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });

  it('should open drawer when a menu meal is clicked', () => {
    const wrapper = mountSetup();
    wrapper.find('Button').simulate('click');
    expect(wrapper.find('Button').length).toEqual(1);
    wrapper.unmount();
  });
});

it('should add extra select input when clicked', () => {
  const wrapper = mountSetup();
  expect(wrapper.find('Dragger').length).toEqual(1);
  expect(wrapper.find('.fa-plus-square').length).toEqual(2);
  const addButton = wrapper.find('.fa-plus-square').first();
  addButton.simulate('click');
  // expect(wrapper.find('.fa-minus.extra-minus').length).toEqual(1);
  const addButton2 = wrapper.find('.fa-plus-square').last();
  addButton2.simulate('click');
  // expect(wrapper.find('.fa-minus.extra-minus').exists()).toEqual(true);
  // expect(wrapper.find('select').length).toEqual(1);
  wrapper.unmount();
});

// it('should add extra select input when clicked', () => {
//   const wrapper = setup();
//   const addButton = wrapper.find('input').at(1);
//   // expect(addButton).toEqual(1);
//   addButton.simulate('change', { target: { value: 'folajimi' } });
//   expect(addButton.prop('value')).toEqual('gfhgh');
//   // expect(wrapper.find('select').length).toEqual(1);
//   wrapper.unmount();
// });