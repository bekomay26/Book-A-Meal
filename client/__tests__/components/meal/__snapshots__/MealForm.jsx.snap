// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Meal Form Component should render meal form component correctly 1`] = `
<div>
  <Form
    as="form"
    className="add-meal-form"
    onSubmit={[MockFunction]}
  >
    <Dragger
      accept="image/*"
      action=""
      listType="picture-card"
      name="filename"
      onChange={[MockFunction]}
      showUploadList={false}
    >
      <div>
        <Icon
          type="plus"
        />
        <img
          alt="meal"
          className="menu-meal-img"
          src="https://res.cloudinary.com/dffiyhgto/image/upload/v1536160768/gfhgh980.jpg"
        />
      </div>
    </Dragger>
    <FormGroup>
      <FormField
        width={10}
      >
        <label>
          Food Title
        </label>
        <Input
          name="title"
          onChange={[MockFunction]}
          placeholder="Meal"
          type="text"
          value="gfhgh"
        >
          <input />
        </Input>
      </FormField>
      <FormField
        width={6}
      >
        <label>
          Price
        </label>
        <Input
          className="meal-update-price-input"
          labelPosition="right"
          max="9995"
          min="50"
          name="price"
          onChange={[MockFunction]}
          placeholder="Amount"
          step="5"
          type="number"
          value={980}
        >
          <Label
            basic={true}
          >
            $
          </Label>
          <input />
          <Label>
            .00
          </Label>
        </Input>
      </FormField>
    </FormGroup>
    <FormTextArea
      as={[Function]}
      control={[Function]}
      label="Description"
      name="description"
      onChange={[MockFunction]}
      placeholder="Tell us more about the food..."
      value=""
    />
    <Grid
      className="add-meal-form-grid"
      columns={2}
      divided={true}
    >
      <GridRow>
        <GridColumn>
          <h4>
            Goes With
          </h4>
          meal
          meal
          <div
            style={
              Object {
                "fontSize": "20px",
                "textAlign": "center",
              }
            }
          >
            <i
              className="far fa-plus-square"
              onClick={[Function]}
              role="button"
            />
          </div>
        </GridColumn>
        <GridColumn>
          <h4>
            Extras
          </h4>
          meal
          meal
          <div
            style={
              Object {
                "fontSize": "20px",
                "textAlign": "center",
              }
            }
          >
            <i
              className="far fa-plus-square"
              onClick={[Function]}
              role="button"
            />
          </div>
        </GridColumn>
      </GridRow>
    </Grid>
    <FormButton
      as={[Function]}
      className="manage-meal-button"
      control={[Function]}
    >
      Update
    </FormButton>
  </Form>
</div>
`;
